topdown/trace.go

```diff
@@ -196,6 +206,24 @@ func formatEventSpaces(event *Event, depth int) int {
 	return depth + 1
 }
 
+func formatLocation(event *Event) string {
+	if event.Op == NoteOp {
+		return fmt.Sprintf("%-19v", "note")
+	}
+
+	location := event.Location
+	if location == nil {
+		// Most likely a trace event
+		return fmt.Sprintf("%-19v", "trace")
+	}
+
+	if location.File == "" {
+		return fmt.Sprintf("%-19v", fmt.Sprintf("%.15v:%v", "buffer", location.Row))
+	}
+
+	return fmt.Sprintf("%-19v", fmt.Sprintf("%.15v:%v", location.File, location.Row))
```

Comment:
It might be a nice addition in the future to have the spacing be a little bit more dynamic to reduce the empty gap on short names and allow for longer names (to some reasonable extent). Nothing we need to change now though.. I think that is something we'll have to play around with by using this to see what works well.

Reply:
Yeah, initially I was thinking the same thing, but decided for know it makes sense to restrict it. It prevents really long file names from messing up the trace output.